### PR TITLE
add odh-model-controller as an independent component

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
## Description

- Add odh-model-controller as an independent component

Note: the existing odh-model-controller `overlay` to modelmesh has been kept as it is to minimize any impact of changes to modelmesh functionality 

## How Has This Been Tested?

- This PR should be tested along with corresponding PRs to [caikit-tgis-serving](https://github.com/opendatahub-io/caikit-tgis-serving/pull/46) and [odh-model-controller](https://github.com/opendatahub-io/odh-model-controller/pull/65)
- To test the new controller : follow the instructions [here](https://github.com/VedantMahabaleshwarkar/caikit-tgis-serving/blob/metrics-controller/demo/kserve/Kserve.md) 

### Testing Instructions with opendatahub-operator 2.0 
- deploy the operator using 
`make deploy -e IMG=quay.io/vedantm/opendatahub-operator:latest -e OPERATOR_NAMESPACE=opendatahub-operator-system` 
- Use the following DSCInit CR 
```
spec:
  applicationsNamespace: opendatahub
  manifestsUri: 'https://github.com/VedantMahabaleshwarkar/odh-manifests/tarball/op2test'
  monitoring:
    enabled: false
    namespace: opendatahub
```
- Use the following DSC
```
spec:
  components:
    codeflare:
      enabled: false
    dashboard:
      enabled: false
    datasciencepipelines:
      enabled: false
    kserve:
      enabled: true
    modelmeshserving:
      enabled: false
    ray:
      enabled: false
    workbenches:
      enabled: false
```
- Verify that `odh-model-controller` deployment is successful with the image `quay.io/vedantm/odh-model-controller:latest` 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: odh-model-controller/base/params.env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+monitoring-namespace=opendatahub
+odh-mm-rest-proxy=quay.io/opendatahub/rest-proxy:v0.10.0
```
Comment (Xaenalt) [inline odh-model-controller/base/params.env:2]: v0.11?

**Review by heyselbi — APPROVED** [review]

**Review by heyselbi — APPROVED** [review]

File: odh-model-controller/base/params.env (hunk 1 above)
Comment (VedantMahabaleshwarkar) [reply odh-model-controller/base/params.env]: Removed

**Comment by zdtsw** [comment]
can we get this one merge to "master" now? 
logic in operator v2 has been updated, since it is missing these changes, it causes failure if enable "kserve"

**Pull request merged**